lib/auth/trustedcluster.go

```diff
@@ -550,10 +580,54 @@ func (a *Server) validateTrustedClusterToken(ctx context.Context, tokenName stri
 	return provisionToken.GetMetadata().Labels, nil
 }
 
-func (a *Server) sendValidateRequestToProxy(host string, validateRequest *authclient.ValidateTrustedClusterRequest) (*authclient.ValidateTrustedClusterResponse, error) {
+func (a *Server) sendValidateRequestToProxy(ctx context.Context, clt *roundtrip.Client, validateRequest *authclient.ValidateTrustedClusterRequest) (*authclient.ValidateTrustedClusterResponse, error) {
+	validateRequestRaw, err := validateRequest.ToRaw()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	out, err := httplib.ConvertResponse(clt.PostJSON(ctx, clt.Endpoint("webapi", "trustedclusters", "validate"), validateRequestRaw))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var validateResponseRaw authclient.ValidateTrustedClusterResponseRaw
+	err = json.Unmarshal(out.Bytes(), &validateResponseRaw)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	validateResponse, err := validateResponseRaw.ToNative()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return validateResponse, nil
+}
+
+// validateTrustedClusterName validates that the trusted cluster resource name
+// matches the cluster name.
+func (a *Server) validateTrustedClusterName(ctx context.Context, clt *roundtrip.Client, trustedCluster types.TrustedCluster) error {
+	out, err := httplib.ConvertResponse(clt.Get(ctx, clt.Endpoint("webapi", "ping"), nil))
```

Comment:
Is there a way to reuse the `api/client/webclient.Find` code?

Also, `find` still has the cluster name so that's preferable to `ping`.